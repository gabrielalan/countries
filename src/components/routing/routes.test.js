import React from 'react';
import renderer from 'react-test-renderer';
import Routes from './routes';

const PassThrough = ({ children }) => children;
const Route = ({ render }) => render({});
const Redirect = ({ children }) => <div className="redirect">{children}</div>;
const Page = () => <span>Page</span>;

describe('routes', () => {
  it('should render pages correctly secured', () => {
    const pages = [
      { path: '/', component: Page },
      { path: '/page', isPrivate: true, component: Page },
    ];

    const comp = <Routes
      Route={Route}
      Router={PassThrough}
      Switch={PassThrough}
      Redirect={Redirect}
      pages={pages} />;

    const tree = renderer.create(comp).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
