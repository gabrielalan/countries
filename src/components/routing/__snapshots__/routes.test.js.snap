// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes should render loading if its loading 1`] = `
<div
  className="s-mt--30"
>
  <div
    className="loading"
  >
    <div
      className="loading__icon"
    >
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;

exports[`routes should render pages correctly secured 1`] = `
Array [
  <span>
    Page
  </span>,
  <div
    className="redirect"
  >
    {"pathname":"/login","state":{}}
     - 
  </div>,
  <div
    className="redirect"
  >
    "/"
     - 
  </div>,
]
`;
